File: skills/invoice-conversion/examples/ubl-to-json.js
#!/usr/bin/env node
// Convert a UBL XML file to Invapi Invoice JSON
// Usage: node ubl-to-json.js <invoice.ubl.xml>
//
// Requires: INVAPI_API_KEY environment variable

const fs = require("fs");

const input = process.argv[2];
if (!input) {
  console.error("Usage: node ubl-to-json.js <invoice.ubl.xml>");
  process.exit(1);
}

const apiKey = process.env.INVAPI_API_KEY;
if (!apiKey) {
  console.error("Error: INVAPI_API_KEY environment variable is not set.");
  process.exit(1);
}

const body = fs.readFileSync(input, "utf-8");

fetch("https://api.invapi.org/api/v1/ubl/json", {
  method: "POST",
  headers: {
    "x-api-key": apiKey,
    "Content-Type": "application/xml",
  },
  body,
})
  .then((res) => res.text())
  .then((text) => console.log(text))
  .catch((err) => {
    console.error(err);
    process.exit(1);
  });

File: skills/invoice-conversion/examples/ubl-to-json.py
#!/usr/bin/env python3
# Convert a UBL XML file to Invapi Invoice JSON
# Usage: python ubl-to-json.py <invoice.ubl.xml>
#
# Requires: INVAPI_API_KEY environment variable, requests library

import os
import sys

import requests

if len(sys.argv) < 2:
    print("Usage: python ubl-to-json.py <invoice.ubl.xml>", file=sys.stderr)
    sys.exit(1)

input_path = sys.argv[1]

api_key = os.environ.get("INVAPI_API_KEY")
if not api_key:
    print("Error: INVAPI_API_KEY environment variable is not set.", file=sys.stderr)
    sys.exit(1)

with open(input_path, "r") as f:
    body = f.read()

resp = requests.post(
    "https://api.invapi.org/api/v1/ubl/json",
    headers={"x-api-key": api_key, "Content-Type": "application/xml"},
    data=body,
)
resp.raise_for_status()
print(resp.text)
